front-end/virtual_pet/lib/pages/pomodoro_page.dart

```diff
@@ -1,30 +1,124 @@
+// ignore_for_file: sized_box_for_whitespace
+
 import 'package:flutter/material.dart';
-import 'package:virtual_pet/widgets/app_bar.dart';
+import 'minutes.dart';
+import 'dart:async';
 
 /// pomodoro page (pomodoro timer)
 class PomodoroPage extends StatefulWidget {
   const PomodoroPage({super.key});
-
   @override
   PomodoroState createState() => PomodoroState();
 }
 
 /// the state of pomodoro page (timer)
 class PomodoroState extends State<PomodoroPage> {
+  int timeLeft = 0;
   static const String routeName = "/PomodoroPage";
+  bool inProgress = false;
+  int lastValue = 0;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  void _startCountDown() {
+    setState(() {
+      inProgress = true;
+      lastValue = timeLeft;
+    });
+    Timer.periodic(const Duration(seconds: 1), (timer) {
+      if (timeLeft > 0) {
+        setState(() {
+          timeLeft--;
+        });
+      } else {
+        setState(() {
+          inProgress = false;
+        });
+        timer.cancel();
+      }
+    });
+  }
+
+  void _stopTimer() {
+    setState(() {
+      timeLeft = 0;
+      inProgress = false;
+    });
+  }
+
+  void _restartTimer() {
+    setState(() {
+      timeLeft = lastValue;
+    });
+    _startCountDown();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
-        /// displays the "Pomodoro Timer" message at the top
-        appBar: CustomAppBar(title: "Pomodoro Timer"),
-        body: Text(
-          "Pomodoro timer coming soon...",
-          style: TextStyle(
-            color: Colors.black,
-            fontSize: 30.0,
+    return Scaffold(
+      // backgroundColor: Colors.grey[900],
+      body: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          // minutes wheel
+          Container(
+            width: 70,
+            child: ListWheelScrollView.useDelegate(
+              onSelectedItemChanged: (value) {
+                if (inProgress == true) {
+                } else {
+                  setState(() {
+                    timeLeft = value;
+                  });
+                }
+              },
+              itemExtent: 50,
+              perspective: 0.005,
+              diameterRatio: 1.2,
+              physics: const FixedExtentScrollPhysics(),
+              childDelegate: ListWheelChildBuilderDelegate(
+                childCount: 60,
+                builder: (context, index) {
+                  return MyMinutes(
+                    mins: index,
+                  );
+                },
+              ),
+            ),
+          ),
+
+          Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              // Text(inProgress.toString()),
```

Comment:
I know this was for testing purposes as we were discussing earlier. Rather than commenting it out, I would recommend uncommenting it and only displaying it if a testing flag is set to true, or removing it entirely